### ISSUE TITLE
[SCA] Bug in rule processing

### ISSUE THREAD

**Issue by JuantAldea** [wazuh/wazuh#3162]
Upon  investigating the SCA reports performed by QA automation we have found out that there are some issues on the rule processing. The problem seems to manifest when the rule is composite and all terms are negated, i.e, of the form `!check1 && !check2`. More research is needed.

It was discovered while testing rule
```
checks:
 - id: 5004
   title: "Ensure separate partition exists for /var/tmp"
   description: "The /var/tmp directory is a world-writable directory used for temporary storage by all users and some applications."
   rationale: "Since the /var/tmp directory is intended to be world-writable, there is a risk of resource exhaustion if it is not bound to a separate partition. In addition, making /var/tmp its own file system allows an administrator to set the noexec option on the mount, making /var/tmp useless for an attacker to install executable code. It would also prevent an attacker from establishing a hardlink to a system setuid program and wait for it to be updated. Once the program was updated, the hardlink would be broken and the attacker would have his own copy of the program. If the program happened to have a security vulnerability, the attacker could continue to exploit the known flaw."
   remediation: "For new installations, during installation create a custom partition setup and specify a separate partition for /var/tmp . For systems that were previously installed, create a new partition and configure /etc/fstab as appropriate."
   compliance:
    - cis_csc: "5.1"
    - cis: "1.1.7"
   condition: all
   rules:
           - 'f:/etc/fstab -> !r:^# && !r:/var/tmp;'
```
and realizing than no matter what, the rule is always reporting pass, despite of having no lines lines containing `/var/tmp` in the fstab.

|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.9 | Manager/Agent | Sources | -- |

**Comment by JuantAldea** [comment]
WIP in https://github.com/wazuh/wazuh/tree/3162-Bug-in-rule-processing

**Comment by chemamartinez** [comment]
Merged: #3179. GJ @JuantAldea and @TJOSERAFAEL !!
